src/bp/nlu-core/entities/microsoft-extractor/index.ts

```diff
@@ -0,0 +1,380 @@
+      'Litre',
+      'Liter',
+      'Piede cubo',
+      'Cubic foot',
+      'Galón',
+      'Centímetro cúbico',
+      'Millilitro',
+      'Pieds cube',
+      'Teaspoon',
+      'Onces',
+      'Pie cúbico',
+      'Milliliter',
+      'Gallone',
+      'Galão',
+      'Mililitro'
+    ],
+    distance: [
+      'Foot',
+      'Pé',
+      'Picometer',
+      'Millimètres',
+      'Kilometer',
+      'Millimetro',
+      'Pie',
+      'Meter',
+      'Micrometer',
+      'Mile',
+      'Pollice',
+      'Light year',
+      'Mètres',
+      'Centimètres',
+      'Millimeter',
+      'Milímetro',
+      'Hectomètre',
+      'Décimètres',
+      'Miglio',
+      'Inch',
+      'Metro',
+      'Quilômetro',
+      'Milha',
+      'Nanometer',
+      'Milla',
+      'Kilómetro',
+      'Pulgada',
+      'Polegada',
+      'Pied',
+      'Kilomètres',
+      'Pouce',
+      'Piede',
+      'Chilometro'
+    ],
+    weight: [
+      'Ounce',
+      'Oncia',
+      'Libra',
+      'Metric ton',
+      'Gallon',
+      'Pound',
+      'Kilogram',
+      'Tonelada',
+      'Tonne métrique',
+      'Ton',
+      'Barrel',
+      'Tonelada métrica',
+      'Onza',
+      'Libbra',
+      'Gram',
+      'Tonnellata',
+      'Onça',
+      'Tonne',
+      'Livre'
+    ],
+    area: [
+      'Mètre carré',
+      'Metro cuadrado',
+      'Square decameter',
+      'Square hectometer',
+      'Square millimeter',
+      'Chilometro quadrato',
+      'Acro',
+      'Quilômetro quadrado',
+      'Square kilometer',
+      'Square meter',
+      'Kilómetro cuadrado',
+      'Square decimeter',
+      'Acre',
+      'Metro quadrato',
+      'Square centimeter',
+      'Metro quadrado',
+      'Hectare',
+      'Kilomètre carré'
+    ],
+    bits: ['Kilobyte', 'Terabyte', 'bit', 'Gigabyte', 'Megabit', 'Megabyte', 'Bit'],
+    speed: [
+      'Miles par heure',
+      'Kilometer per hour',
+      'Milla por hora',
+      'Meter per second',
+      "Miglia all'ora",
+      'Mile per hour',
+      'Milha por hora'
+    ]
+  }
+
+  private getDucklingUnitMappings = unit => {
+    for (const [type, subtypes] of Object.entries(this.ducklingUnitMapping)) {
+      if (subtypes.includes(unit)) {
+        return type
+      }
+    }
+  }
+
+  private ducklingDateMappings = {
+    // https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-reference-prebuilt-datetimev2?tabs=1-3%2C2-1%2C3-1%2C4-1%2C5-1%2C6-1
+    'datetimeV2.datetimerange': 'duration',
+    'datetimeV2.daterange': 'duration',
+    'datetimeV2.timerange': 'duration',
+    'datetimeV2.duration': 'duration',
+    'datetimeV2.date': 'time',
+    'datetimeV2.datetime': 'time',
+    'datetimeV2.time': 'time'
+  }
+
+  private mapDucklingDates(entity): EntityExtractionResult {
+    const newEntity = { ...entity }
+    const newType = this.ducklingDateMappings[entity.type]
+    newEntity.type = newType
+    newEntity.entityId = `system.${newType}`
+
+    switch (newType) {
+      case 'duration':
```

Comment:
As you said, I believe currently in duckling we only return the start as a 'time' entity. If this is true, juste do the same thing here and we'll make sure to comeback in another PR to give more support for intervals.

I don't feel safe with this '__' string logic.